docs/source/user-guide/vision/quickstarts/plot_detection_tutorial.py

```diff
@@ -224,114 +173,131 @@ def show(imgs):
+    range [0, 255] in a uint8 format.
+    """
+    inp = torch.stack(list(batch[0])).cpu().detach().numpy().transpose((0, 2, 3, 1))
+    mean = [0.485, 0.456, 0.406]
+    std = [0.229, 0.224, 0.225]
+    # Un-normalize the images
+    inp = std * inp + mean
+    inp = np.clip(inp, 0, 1)
+    return inp * 255
+
+def batch_to_labels(batch):
+    """
+    Convert a batch of data to labels in the expected format. The expected format is an iterator of arrays, each array
+    corresponding to a sample. Each array element is in a shape of [B, 5], where B is the number of bboxes
+    in the image, and each bounding box is in the structure of [class_id, x, y, w, h].
+    """
+    tensor_annotations = batch[1]
+    label = []
+    for annotation in tensor_annotations:
+        if len(annotation["boxes"]):
+            bbox = torch.stack(annotation["boxes"])
+            # Convert the Pascal VOC xyxy format to xywh format
+            bbox[:, 2:] = bbox[:, 2:] - bbox[:, :2]
+            # The label shape is [class_id, x, y, w, h]
+            label.append(
+                torch.concat([torch.stack(annotation["labels"]).reshape((-1, 1)), bbox], dim=1)
+            )
+        else:
+            # If it's an empty image, we need to add an empty label
+            label.append(torch.tensor([]))
+    return label
+
+def infer_on_batch(batch, model, device):
+    """
+    Returns the predictions for a batch of data. The expected format is an iterator of arrays, each array
+    corresponding to a sample. Each array element is in a shape of [B, 6], where B is the number of bboxes in the
+    predictions, and each bounding box is in the structure of [x, y, w, h, score, class_id].
+    """
+    nm_thrs = 0.2
+    score_thrs = 0.7
+    imgs = list(img.to(device) for img in batch[0])
+    # Getting the predictions of the model on the batch
+    with torch.no_grad():
+        preds = model(imgs)
+    processed_pred = []
+    for pred in preds:
+        # Performoing non-maximum suppression on the detections
+        keep_boxes = torchvision.ops.nms(pred['boxes'], pred['scores'], nm_thrs)
+        score_filter = pred['scores'][keep_boxes] > score_thrs
+
+        # get the filtered result
+        test_boxes = pred['boxes'][keep_boxes][score_filter].reshape((-1, 4))
+        test_boxes[:, 2:] = test_boxes[:, 2:] - test_boxes[:, :2]  # xyxy to xywh
+        test_labels = pred['labels'][keep_boxes][score_filter]
+        test_scores = pred['scores'][keep_boxes][score_filter]
+
+        processed_pred.append(
+            torch.concat([test_boxes, test_scores.reshape((-1, 1)), test_labels.reshape((-1, 1))], dim=1))
+    return processed_pred
 
 #%%
-# After defining the task class, we can validate it by running the following code:
+# Now we'll create the collate function that will be used by the DataLoader:
 
+def deepchecks_collate_fn(batch):
+    """Return a batch of images, labels and predictions in the expected format."""
```

Comment:
I think its worth extending the discussion here about the purpose of the collate function and how to input 'batch' looks like

Reply:
Done, tell me WYT

Reply:
Looks good, maybe just add that the batch argument is a list of (image,label) tuples (can even be inside the function itself)